### PR TITLE
added course progress bar & completion count

### PR DESCRIPTION
**Pull request by yashnerkar**
### **Before** :
image

### **After** : 
image

### CODE REVIEW

File: src/utils/types/dist/Courses.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+"use strict";
+exports.__esModule = true;
```
Comment (Mehul2410) [inline src/utils/types/dist/Courses.js:2]: delete this file

File: src/utils/types/Courses.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export interface CourseProps {
   course: Course;
   user: any;
+  videoProgress:Number;
```
Comment (Mehul2410) [inline src/utils/types/Courses.ts:4]: type is wrong
it should be number

`videoProgress:number;`

**Comment by Mehul2410** [comment]
delete all dist folders and check why the dist folders are created and resolve this issue

File: src/features/courses/SingleCourse.tsx [hunk 3]
```diff
@@ -40,7 +56,20 @@ function SingleCourse({ course, user }: CourseProps) {
         <div className={styles.courseContent}>
           <h2>{course.title}</h2>
           <p>{course.description}</p>
-        </div>
+        {courseProgress!==undefined && (
```
Comment (Mehul2410) [inline src/features/courses/SingleCourse.tsx:59]: Let's try to use common preetier format i'll share my vscode settings config for preetier

File: src/features/courses/Playlist.tsx [hunk 4]
```diff
@@ -84,10 +84,10 @@ const Playlist = ({ playlist, user, videoProgress }: Props) => {
         }}
       >
         <button className={`primaryButton`}>
-          {videoHistory?.slug !== "" ? "Get Started" : "Continue"}
+          {videoHistory?.slug !== "" ? "Get Started " : "Continue"}
```
Comment (Mehul2410) [inline src/features/courses/Playlist.tsx:87]: Remove space after getting started

File: pages/learn/index.tsx [hunk 5]
```diff
@@ -6,12 +6,19 @@ import { getAllCourses } from "../../src/utils/contentful/courses";
...
+import { CourseProps } from "../../src/utils/types/Courses";
+import { getAllvideoProgresssByUser } from "../../src/utils/api/videoProgressTracking";
+
 
 type Props = {
   allCourses: Array<Course>;
 };
 
-function Learn({ allCourses }: Props) {
+  // const [courseProgress,setCourseProgress] = React.useState(0);
```
Comment (Mehul2410) [inline pages/learn/index.tsx:18]: remove the commented code

File: pages/learn/index.tsx [hunk 6]
```diff
@@ -24,8 +31,19 @@ function Learn({ allCourses }: Props) {
...
       });
     }
+    
   }, [user]);
-  return (
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      const data = await getAllvideoProgresssByUser();
+      console.log(data);
```
Comment (Mehul2410) [inline pages/learn/index.tsx:40]: remove console.log

File: pages/learn/index.tsx [hunk 7]
```diff
@@ -24,8 +31,19 @@ function Learn({ allCourses }: Props) {
...
+      // setCourseProgress(data.length)
+    };
+    fetchData();
+  }, []);
+  console.log(allCourses);
```
Comment (Mehul2410) [inline pages/learn/index.tsx:45]: remove console.log

File: pages/learn/index.tsx [hunk 8]
```diff
@@ -24,8 +31,19 @@ function Learn({ allCourses }: Props) {
...
     }
+    
   }, [user]);
-  return (
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      const data = await getAllvideoProgresssByUser();
+      console.log(data);
+      // setCourseProgress(data.length)
```
Comment (Mehul2410) [inline pages/learn/index.tsx:41]: delete this comment

File: src/utils/types/dist/Courses.js (hunk 1 above)
Comment (yashnerkar) [reply src/utils/types/dist/Courses.js:2]: deleted

File: src/utils/types/Courses.ts (hunk 2 above)
Comment (yashnerkar) [reply src/utils/types/Courses.ts:4]: changed it!

File: src/features/courses/Playlist.tsx (hunk 4 above)
Comment (yashnerkar) [reply src/features/courses/Playlist.tsx:87]: removed space

File: pages/learn/index.tsx (hunk 6 above)
Comment (yashnerkar) [reply pages/learn/index.tsx:40]: removed

File: pages/learn/index.tsx (hunk 7 above)
Comment (yashnerkar) [reply pages/learn/index.tsx:45]: removed!

**Pull request merged**